tina/collectionsSchema/navigationBar.tsx

```diff
@@ -0,0 +1,134 @@
+    },
+    {
+      name: 'navItem',
+      label: 'Navigation Item',
+      list: true,
+      type: 'object',
+      templates: [
+        {
+          name: 'stringItem',
+          label: 'String Item',
+          ui: {
+            itemProps: (item) => {
+              return { label: '📄' + item?.label };
+            },
+          },
+          fields: [
+            {
+              name: 'label',
+              label: 'Label',
+              type: 'string',
+            },
+            {
+              name: 'href',
+              label: 'href',
+              type: 'string',
+            },
+          ],
+        },
+        {
+          name: 'groupOfStringItems',
+          label: 'Group Of String Items',
+          list: true,
+          type: 'object',
+          ui: {
+            itemProps: (item) => {
+              return { label: '📕' + item?.label };
+            },
+          },
+          fields: [
+            {
+              name: 'label',
+              label: 'Label',
+              type: 'string',
+            },
+            {
+              name: 'items',
+              label: 'Page or Submenu',
+              type: 'object',
+              list: true,
+              ui: {
+                itemProps: (item) => {
+                  return { label: item?.label };
+                },
+              },
+              fields: [
+                {
+                  name: 'label',
+                  label: 'Label',
+                  type: 'string',
+                },
+                {
+                  name: 'href',
+                  label: 'href',
+                  type: 'string',
+                },
+              ],
+            },
+          ],
+        },
+        //Note we are duplicating the ModalButton template from /components/blocks/ModalButton.template.ts but this is
+        //because this file is generated before hand and i still want to keep the ModalButton coupled into the components dir too
+
+        {
+          label: 'Modal Button',
+          name: 'modalButton',
+          type: 'object',
+          list: true,
+          ui: {
+            itemProps: (item) => {
+              return { label: '🚨' + item?.label };
+            },
+            defaultItem: {
+              variant: 'default',
+              label: 'Secondary Action',
+              icon: false,
+              size: 'medium',
+            },
+          },
+          fields: [
+            { name: 'label', label: 'Label', type: 'string' },
+            {
+              name: 'color',
+              label: 'Color',
+              type: 'string',
+              options: [
+                { value: 'default', label: 'Seafoam' },
+                { value: 'blue', label: 'Blue' },
+                { value: 'orange', label: 'Orange' },
+                { value: 'white', label: 'White' },
+                { value: 'ghost', label: 'Ghost' },
+                { value: 'command', label: 'Command' },
+              ],
+            },
```

Comment:
As per a previous comment, lets try out the color picker.
https://tina.io/docs/reference/toolkit/fields/color/#definition

Reply:
I dont agree with using the color picker in this instance, as the options provided in the template are not just a background styling. If it were just a straight color (i.e background color) then i would use the color picker. 

In `button.tsx` they decide this styling of the `<button/>` object such:

`className={`${baseClasses} ${
        colorClasses[color] ? colorClasses[color] : colorClasses['seafoam']`

which looks like 

const colorClasses = {
  seafoam:
    raisedButtonClasses +
    ' text-orange-600 hover:text-orange-500 border border-seafoam-150 bg-gradient-to-br from-seafoam-50 to-seafoam-150',
  blue:
    raisedButtonClasses +
    ' text-white hover:text-gray-50 border border-blue-400 bg-gradient-to-br from-blue-300 via-blue-400 to-blue-600',
  orange:
    raisedButtonClasses +
    ' text-white hover:text-gray-50 border border-orange-600 bg-gradient-to-br from-orange-400 to-orange-600',
  white:
    raisedButtonClasses +
    ' text-orange-500 hover:text-orange-400 border border-gray-100/60 bg-gradient-to-br from-white to-gray-50',
  ghost: 'text-orange-500 hover:text-orange-400',
  orangeWithBorder: 'text-orange-500 hover:text-orange-400 border border-orange-500 bg-white',
}

Reply:
understood - thanks